extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1061,4 +1062,54 @@ public Locale getLanguageInfo() {
     public String getSupportInfo() {
         return "";
     }
+
+    @Nonnull
+    @Override
+    public List<StreamSegment> getStreamSegments() {
+        final ArrayList<StreamSegment> segments = new ArrayList<>();
+        try {
+            final JsonArray panels = initialData.getArray("engagementPanels");
+            JsonArray segmentArray = null;
+
+            // Search for correct JsonObject containing the data => panelIdentifier == engagement-panel-macro-markers
+            for (int i = 0; i < panels.size(); i++) {
+                if (panels.getObject(i).getObject("engagementPanelSectionListRenderer")
+                        .getString("panelIdentifier").equals("engagement-panel-macro-markers")) {
+                    segmentArray = panels.getObject(i).getObject("engagementPanelSectionListRenderer")
+                            .getObject("content").getObject("macroMarkersListRenderer").getArray("contents");
+                    break;
+                }
+            }
+
+            if (segmentArray == null) {
+                throw new Exception("Couldn't find valid segments");
```

Comment:
Throwing Exceptions is expensive, because generating the tree of calls can take some time. Please invert the if statement as you are ignoring the thrown Exception anyway.

Reply:
Alright, done.